tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/macro/AbstractReusableActiveAnnotationTests.xtend

```diff
@@ -3363,6 +3363,87 @@ abstract class AbstractReusableActiveAnnotationTests {
 		]
 	}
 	
+	@Test def void testMoveExpressionToTypeParameterDeclarator() {
+		assertProcessing(
+			'myannotation/AbstractAnnotation.xtend' -> '''
+				package myannotation
+				
+				import java.lang.annotation.ElementType
+				import java.lang.annotation.Target
+				import org.eclipse.xtend.lib.macro.AbstractClassProcessor
+				import org.eclipse.xtend.lib.macro.Active
+				import org.eclipse.xtend.lib.macro.TransformationContext
+				import org.eclipse.xtend.lib.macro.declaration.MethodDeclaration
+				import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration
+				
+				@Target(ElementType.TYPE)
+				@Active(SomeAnnotationProcessor)
+				annotation SomeAnnotation {
+				}
+				
+				class SomeAnnotationProcessor extends AbstractClassProcessor {
+				
+					static def copyMethod(MutableClassDeclaration clazz, MethodDeclaration method, extension TransformationContext context) {
+				
+						val methodDeclaration = clazz.addMethod(method.simpleName) [
+							it.abstract = method.abstract
+							it.docComment = method.docComment
+							it.visibility = method.visibility
+							it.synchronized = method.synchronized
+							it.deprecated = method.deprecated
+							it.static = method.static
+							it.exceptions = method.exceptions
+							it.varArgs = method.varArgs
+							it.strictFloatingPoint = method.strictFloatingPoint
+						]
+				
+						for (t : method.typeParameters) {
```

Comment:
If i remove copying of type parameters then T will be resolved to the type parameter T of the original method and we will get quite confusing error: 'Cannot make a static reference to the non-static type T'. I wonder should the resolution of T fail, since the body is moved to another method?